canvas-image.js

```diff
@@ -92,22 +123,193 @@ function handleFileSelect (e) {
+		if (e.altKey) {
+			// alt + numbers
+			switch(e.which) {
+			case "1".charCodeAt(0):
+				document.getElementById('rdo_grayscale_average').click();
+				break;
+			case "2".charCodeAt(0):
+				document.getElementById('rdo_grayscale_decomposition').click();
+				break;
+			case "3".charCodeAt(0):
+				document.getElementById('rdo_grayscale_desaturation').click();
+				break;
+			case "4".charCodeAt(0):
+				document.getElementById('rdo_grayscale_luminance').click();
+				break;
+			case "5".charCodeAt(0):
+				document.getElementById('rdo_grayscale_shades').click();
+				break;
+			case "6".charCodeAt(0):
+				document.getElementById('rdo_grayscale_singlechannel').click();
+				break;
+			case "0".charCodeAt(0):
+				document.getElementById('rdo_grayscale_disable').click();
+				break;
+			default:
+				// nothing
+			}
+		} else {
+			// numbers
+			switch(e.which) {
+			case "i".charCodeAt(0):
+			case "I".charCodeAt(0):
+				document.getElementById('fileSelect').click();
+				break;
+
+			case "s".charCodeAt(0):
+			case "S".charCodeAt(0):
+				document.getElementById('rng_scale').focus();
+				break;
+
+			case "1".charCodeAt(0):
+				document.getElementById('rdo_dither_atkinson').click();
+				break;
+			case "2".charCodeAt(0):
+				document.getElementById('rdo_dither_burkes').click();
+				break;
+			case "3".charCodeAt(0):
+				document.getElementById('rdo_dither_falsefloydsteinberg').click();
+				break;
+			case "4".charCodeAt(0):
+				document.getElementById('rdo_dither_floydsteinberg').click();
+				break;
+			case "5".charCodeAt(0):
+				document.getElementById('rdo_dither_jarvisjudiceninke').click();
+				break;
+			case "6".charCodeAt(0):
+				document.getElementById('rdo_dither_sierra').click();
+				break;
+			case "7".charCodeAt(0):
+				document.getElementById('rdo_dither_sierratworow').click();
+				break;
+			case "8".charCodeAt(0):
+				document.getElementById('rdo_dither_sierralite').click();
+				break;
+			case "9".charCodeAt(0):
+				document.getElementById('rdo_dither_stucki').click();
+				break;
+			case "t".charCodeAt(0):
+			case "T".charCodeAt(0):
+				document.getElementById('rdo_dither_threshold').click();
+				break;
+			case "0".charCodeAt(0):
+				document.getElementById('rdo_dither_disable').click();
+				break;
+
+			case "r".charCodeAt(0):
+			case "R".charCodeAt(0):
+				document.getElementById('chk_replace_colours').click();
+				break;
+
+			case "p".charCodeAt(0):
+			case "P".charCodeAt(0):
+				document.getElementById('rdo_format_png').click();
+				break;
+			case "g".charCodeAt(0):
+			case "G".charCodeAt(0):
+				document.getElementById('rdo_format_gif').click();
+				break;
+			case "b".charCodeAt(0):
+			case "B".charCodeAt(0):
+				document.getElementById('rdo_format_bmp').click();
+				break;
+			
+			default:
+				// nothing
+			}
+		}
+
+		// redraw on any key press
+		draw();
+	};
+
 }
 
-window.onload = setup;
\ No newline at end of file
+// main setup function now called in html
```

Comment:
We should probably key this off [`DOMContentLoaded`](https://developer.mozilla.org/en-US/docs/Web/API/Window/DOMContentLoaded_event) here instead, and can remove the call from the HTML!

```suggestion
window.addEventListener('DOMContentLoaded', setup);
```

Reply:
Absolutely.